File: Core/Stateflows.Common/Events/BehaviorInfo.cs
using System;
using Newtonsoft.Json;
using System.Collections.Generic;

namespace Stateflows.Common
{
    [Retain, StrictOwnership]
    public class BehaviorInfo
    {
        public BehaviorId Id { get; set; }
        
        public BehaviorStatus BehaviorStatus { get; set; }
        
        public string BehaviorStatusText => Enum.GetName(typeof(BehaviorStatus), BehaviorStatus);

        [JsonProperty(TypeNameHandling = TypeNameHandling.None)]
        public IEnumerable<string> ExpectedEvents { get; set; }

        public Dictionary<string, object> Metadata { get; set; } = new();
    }
}

File: Transport/@stateflows/common/src/events/behavior-info.ts
import { StateflowsEvent } from "../decorators/stateflows-event";
import { BehaviorStatus } from "../enums/behavior-status";
import { Event } from "./event";

@StateflowsEvent("Stateflows.Common.Events.BehaviorInfo, Stateflows.Common")
export class BehaviorInfo extends Event {
    public behaviorStatus: BehaviorStatus;
    public expectedEvents: Array<string>;
}